oauth2/oauth2-core/src/main/java/org/springframework/security/oauth2/core/endpoint/OAuth2AuthorizationRequest.java

```diff
@@ -44,6 +43,7 @@
  * @see OAuth2AuthorizationResponseType
  * @see <a target="_blank" href="https://tools.ietf.org/html/rfc6749#section-4.1.1">Section 4.1.1 Authorization Code Grant Request</a>
  * @see <a target="_blank" href="https://tools.ietf.org/html/rfc6749#section-4.2.1">Section 4.2.1 Implicit Grant Request</a>
+ * @see <a target="_blank" href="https://tools.ietf.org/html/rfc7636#section-4.3">Section 4.3 Client Sends the Code Challenge with the Authorization Request</a>
  */
 public final class OAuth2AuthorizationRequest implements Serializable {
```

Comment:
I think it makes sense to allow this to be extended so that there is a specific implementation that is used for supporting pkce. This makes sense because we don't want to have a request that supports all extensions making it confusing for users who only need a simple request.

Reply:
I agree that it makes sense to allow `OAuth2AuthorizationRequest` to be extended but don't feel that it's required at this moment. PKCE supplies 2 parameters and is quite simple. I think this logic can be default logic in `DefaultOAuth2AuthorizationRequestResolver`. See [comment](https://github.com/spring-projects/spring-security/pull/6485#discussion_r253268516)